### ISSUE TITLE
LEt MultiIndex ax

### ISSUE THREAD

**Issue by topper-123** [pandas-dev/pandas#23887]
#### Code Sample, a copy-pastable example if possible

```python
# Your code here

```
#### Problem description

[this should explain **why** the current behaviour is a problem and why the expected output is a better solution.]

**Note**: We receive a lot of issues on our GitHub tracker, so it is very possible that your issue has been posted before. Please check first before submitting so that we do not have to handle and close duplicates!

**Note**: Many problems can be resolved by simply upgrading `pandas` to the latest version. Before submitting, please check if that solution works for you. If possible, you may want to check if `master` addresses this issue, but that is not necessary.

For documentation-related issues, you can check the latest versions of the docs on `master` here:

https://pandas-docs.github.io/pandas-docs-travis/

If the issue has not been resolved there, go ahead and file it in the issue tracker.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

</details>

**Comment by ArtinSarraf** [comment]
The arrays/dictionary initialization would be covered in the from_data proposal from this PR: #23141. Although pd.MultiIndex.from_arrays/from_tuples would already cover list-like input.

@jreback / @toobaz / @TomAugspurger - does this add any merit to from_data over the dedicated from_frame?

**Comment by gfyoung** [comment]
I'm confused here...don't we already have `MultiIndex.from_tuples` ?

**Comment by jreback** [comment]
the proposal to make the default constructor to this

**Comment by gfyoung** [comment]
@topper-123 : Yep, that makes sense.  +1 from me.

**Comment by shoyer** [comment]
I don't think this makes sense. The default `pd.MultiIndex` constructor intentionally mirrors the under-lying data-structure (in terms of `levels` and `codes`). It's OK if users don't use the constructor directly -- we can aways make convenience functions for that (which indeed already exist). We could even make a new "try everything" converter function `pd.to_multiindex()`, though I'm not sure how you could reliably distinguish `from_tuples` from `from_arrays`.

There is no iron-clad rule that a class's repr needs to use its constructor directly -- we could make the repr show `MultiIndex.from_tuples(...)` if desired. The situation is pretty similar to that for `np.ndarray` (with its low-level constructor) and `np.array` (which is used in the repr).

If you did go ahead with this, what would you do with the existing constructor? Would you delegate the existing constructor to another method, or would you simply overload the first argument `levels` to check for a list of tuples? This sort of heavy overloading makes code harder to understand and error check (either manually or with static checks like mypy/pylint).
